ext/standard/crc32.c

```diff
@@ -28,13 +55,60 @@ PHP_NAMED_FUNCTION(php_if_crc32)
 	size_t nr;
 	uint32_t crcinit = 0;
 	register uint32_t crc;
+#if defined(__aarch64__)
+	int16_t *p2;
+	int32_t *p4;
+	int64_t *p8;
+#endif
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
 		Z_PARAM_STRING(p, nr)
 	ZEND_PARSE_PARAMETERS_END();
 
 	crc = crcinit^0xFFFFFFFF;
 
+#if defined(__aarch64__)
+	if (has_crc32_insn()) {
```

Comment:
It probably, may make sense to reuse ZEND_INTRIN_... mechanism to use link-time ifunc resolver.

Reply:
Thanks for the feedback: I will amend the patch and resubmit for review.

Reply:
I would prefer not to introduce ifunc usage on aarch64. Our existing usages have already caused a lot of trouble for little gain.

Reply:
Then, it's easier to keep the patch as is. I can't test it, but speed improvement looks promising.